### PR TITLE
#1280 - Index for fullname DB Search

### PR DESCRIPTION
**Pull request by sh16011993**
Created a GIN index for fullname DB Search. This improves the retrieval speed of the records for the following screens if a search is made:

- Institution **Manage User** section

image

- Institution **Confirm Enrollment** section

image

- Ministry **Student Requests - Appeals** section

image

- Ministry **Student Requests - Exceptions** section

image

### CODE REVIEW

File: sources/packages/backend/apps/db-migrations/src/sql/User/Add-index-first-name-last-name.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- # pg_trgm extension to enable gin index.
+CREATE EXTENSION pg_trgm;
+
+-- # gin index on the concatenated first_name and last_name column.
+CREATE INDEX first_name_last_name_gin_idx ON sims.users USING gin((first_name || ' ' || last_name) gin_trgm_ops);
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/db-migrations/src/sql/User/Add-index-first-name-last-name.sql:5]: Please check others `CREATE INDEX` SQL files. I believe that we have been avoiding using the suffix "idx".

**Review by guru-aot — APPROVED** [review]
Nice work @sh16011993. Please work on the other devs comments.

File: sources/packages/backend/apps/api/src/services/application-exception/application-exception.service.ts [hunk 2]
```diff
@@ -214,7 +214,7 @@ export class ApplicationExceptionService extends RecordDataModelService<Applicat
         .andWhere(
           new Brackets((qb) => {
             qb.where(
-              "CONCAT(user.firstName,' ', user.lastName) Ilike :searchCriteria",
+              "(user.firstName || ' ' || user.lastName) Ilike :searchCriteria",
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/services/application-exception/application-exception.service.ts:217]: We can have a simple util method to create this string in a centralized way
```ts
export function getUserFulNameLikeSearch(userTableAlies = "user", searchCriteriaParameteName = "searchCriteria"){
    return `(${userTableAlies }.firstName || ' ' || ${userTableAlies }.lastName) ILIKE :${searchCriteriaParameteName }`;
}
```

File: sources/packages/backend/apps/db-migrations/src/sql/User/Add-index-first-name-last-name.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+-- # pg_trgm extension to enable gin index.
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/db-migrations/src/sql/User/Add-index-first-name-last-name.sql:1]: Is there a reason for the "#" at the beginning of all comments?

**Review by andrewsignori-aot — COMMENTED** [review]
Great work doing the research and applying the solution. Please take look at the minor comment.

File: sources/packages/backend/apps/api/src/services/application-exception/application-exception.service.ts (hunk 2 above)
Comment (sh16011993) [reply sources/packages/backend/apps/api/src/services/application-exception/application-exception.service.ts:217]: Thanks for suggesting this change.

**Review by dheepak-aot — APPROVED** [review]
Great analysis and work @sh16011993 . 👏

As @andrewsignori-aot having the centralized method to have the full name search will be good.

File: sources/packages/backend/apps/db-migrations/src/sql/User/Add-index-first-name-last-name.sql (hunk 3 above)
Comment (sh16011993) [reply sources/packages/backend/apps/db-migrations/src/sql/User/Add-index-first-name-last-name.sql:1]: I just followed the pattern from other sql files. I can remove it. Only `--` is required for comment

Comment (sh16011993) [reply sources/packages/backend/apps/db-migrations/src/sql/User/Add-index-first-name-last-name.sql]: Removed the `#`

**Review by andrepestana-aot — APPROVED** [review]
Great job! Thanks for the explanation about the indexes!

**Pull request merged**